testsuites/rio/src/main/java/org/eclipse/rdf4j/rio/AbstractParserHandlingTest.java

```diff
@@ -948,4 +951,15 @@ private final Model getTestModel(String languageValue, String languageTag) {
 				vf.createLiteral(languageValue, languageTag)));
 		return result;
 	}
+
+	@Test(expected = RDFParseException.class)
```

Comment:
There need to be other tests where a valid document stream is sent in and parsing succeeds even with the rdfHandler being set to a non-null value.